code/modules/mob/living/carbon/xenomorph/castes/spiderling/spiderling.dm

```diff
@@ -15,43 +21,67 @@
 	allow_pass_flags = PASS_XENO
 	pass_flags = PASS_XENO|PASS_LOW_STRUCTURE
 	density = FALSE
+	// The widow that this spiderling belongs to
+	var/mob/living/carbon/xenomorph/spidermother
+	// What sprite state this - normal, enraged, guarding? Used for update_icons()
+	var/spiderling_state = SPIDERLING_NORMAL
+
+/mob/living/carbon/xenomorph/spiderling/Initialize(mapload, mob/living/carbon/xenomorph/mother)
+	. = ..()
+	AddComponent(/datum/component/ai_controller, /datum/ai_behavior/spiderling, mother)
+	spidermother = mother
 
-/mob/living/carbon/xenomorph/spiderling/Initialize(mapload, mob/living/carbon/xenomorph/spidermother)
+/mob/living/carbon/xenomorph/spiderling/update_icons(state_change = TRUE)
 	. = ..()
-	AddComponent(/datum/component/ai_controller, /datum/ai_behavior/spiderling, spidermother)
+	if(state_change)
+		if(spiderling_state == SPIDERLING_ENRAGED)
+			icon_state = "[icon_state] Enraged"
+		if(spiderling_state == SPIDERLING_GUARDING)
+			icon_state = "[icon_state] Guarding"
 
 /mob/living/carbon/xenomorph/spiderling/on_death()
 	///We QDEL them as cleanup and preventing them from being sold
 	QDEL_IN(src, TIME_TO_DISSOLVE)
 	return ..()
 
+//If we're covering our widow, any clicks should be transferred to them
+/mob/living/carbon/xenomorph/spiderling/Click(location, control, params)
```

Comment:
Click shouldn't be overriden. Try `attack_alien` instead

Reply:
attack_alien won't cover for ability usage, e.g heals. afaik, the only way to cover for that is through either a catch-all click() or overriding more specific stuff like right/middle/alt clicks() - would that be acceptable? are there any other, cleaner methods?